### PR TITLE
feat: remove `ChainTypes`

### PR DESCRIPTION
**Pull request by 0xApotheosis**
**WIP**

- Move `chain-adapter` types out of the `shapeshiftoss/types` package and into `shapeshiftoss/types`. This is needed to prevent circular dependencies
- Added nominal typing to CAIP types
- Replace `ChainTypes` with `ChainNamespace`
- Make `chain` and `network` Asset properties optional, and remove respective data from mocks
- Replace `SupportedChainId` chain-adapter generic type with something more generic so that any valid `ChainId` can be used
- Remove `ChainAdapterManager` - it's not actually needed

**TODO**

- De-structure `chain-adapter` types into common directories
- Remove "chain type" vernacular from affected code
- Update READMEs
- Address FIXMEs

**Reviewers**

Key changed files:

- `packages/chain-adapters/src/types/index.ts` - `ChainTypes` changed to `SupportedChainId`
- `packages/chain-adapters/src/ChainAdapterManager.ts` - switch on `SupportedChainId`.

### CODE REVIEW

File: packages/chain-adapters/src/ChainAdapterManager.ts [hunk 1]
```diff
@@ -81,15 +99,19 @@ export class ChainAdapterManager {
...
     if (typeof chain !== 'string' || typeof factory !== 'function') {
       throw new Error('Parameter validation error')
     }
     this.supported.set(chain, factory)
   }
 
-  removeChain<T extends ChainTypes>(chain: T): void {
-    if (!Object.values(ChainTypes).includes(chain)) {
+  removeChain<T extends SupportedChainId>(chain: T): void {
+    // FIXME: This is gross
```
Comment (kaladinlight) [inline packages/chain-adapters/src/ChainAdapterManager.ts:113]: do we really even need this? T is already constrained to `SupportedChainId` and seem redundant (and gross as mentioned) to do this additional check

Comment (gomesalexandre) [reply packages/chain-adapters/src/ChainAdapterManager.ts:113]: Yeah this logic should live in a type guard, but if you move it to a type guard, you realize it should always return true according to the typings

File: packages/types/src/base.ts [hunk 2]
```diff
@@ -15,14 +15,14 @@ export enum ChainTypes {
   Osmosis = 'osmosis'
 }
 
-const supportedChainIds = [
-  'eip155:1',
-  'bip122:000000000019d6689c085ae165831e93',
-  'cosmos:cosmoshub-4',
-  'cosmos:osmosis-1'
-] as const
+export enum SupportedChainIds {
```
Comment (gomesalexandre) [inline packages/types/src/base.ts:18]: I don't think `supportedChainIds`/`SupportedChainIds` is necessarily bad for now, plus we now have them indexed with a respective enum key.

File: packages/chain-adapters/src/ChainAdapterManager.ts (hunk 1 above)
Comment (0xApotheosis) [reply packages/chain-adapters/src/ChainAdapterManager.ts]: We killed `ChainAdapterManager` entirely - problem solved!

**Pull request closed without merging**